  {% extends "layout.html" %}

  {% block content %}

  <!-- search box -->
  <div id="search-container">
    <h2>Cas protein search</h2>
    <form action="{{ url_for("gfg")}}" method="POST">
      <label for="search-box"></label><textarea
          id="search-box" name="search-box" rows="3" cols="50"
          placeholder="Available search formats:&#10; 1- FASTA-formatted sequence&#10; 2- Word search (e.g. Cas name or accession)&#10; 3- Or simply hit 'Search' for the current full list of entries"></textarea>
      <input type="submit" value="Search" name="seq-input">
      <!--input type="file" name="file-input"-->
    </form>
  </div>

  <!-- CasPEDIA Summary -->
  <div id="head-summary" class="summary">
  <div class="abstract">
    <h2><b>WHAT IS CasPEDIA?</b></h2>
    <h3>CasPEDIA, or the Cas Protein Effector Database of Information and Assessment, is an encyclopedia of Class 2 CRISPR systems, presented in wiki format. Wiki entries provide comprehensive descriptions of enzyme activities, structures and sequences, complete with a literature review covering each nuclease's discovery, experimental considerations and applications.</h3>

    <p>CasPEDIA introduces CasIDs, which organize Cas enzymes by their functional properties. Inspired by the <a href="https://doi.org/10.1093/nar/28.1.304" target="_blank">ENZYME Classification (E.C.) system</a>, CasID nomenclature is composed of three digits, described <a href="#section_multiplx">below</a>, that concisely describe effector properties and enable rapid identification of appropriate enzymes for a researcher's use case. CasIDs can be used to navigate the website through direct search in the search bar and using our <a href="http://caspedia.org/tool_finder.html" target="_blank">Tool Finder</a>.</p>

    <p>Please visit the <a href="http://caspedia.org/faq.html" target="_blank">FAQ</a> section for directions on navigating CasPEDIA and information about becoming a curator. Additional details about CasPEDIA are available in our <a href = "https://doi.org/10.32942/X2C31F" target="_blank">preprint</a>. Individuals interested in learning CRISPR fundamentals are encouraged to read <a href="https://innovativegenomics.org/crisprpedia/" target="_blank">CRISPRpedia</a>, created by The Innovative Genomics Institute.</p>

    <h2><b>CITATION</b></h2>
    <h3>Currently, in preprint</h3>
    <p><b>CasPEDIA Database: A Functional Classification System for Class 2 CRISPR-Cas Enzymes</b><br><i> Adler, B. A.*, Trinidad, M. I.*, Bellieny-Rabelo, D., Zhang, E., Karp, H. M., Skopintsev, P., ... & Doudna, J. A. (2023). Nucleic Acids Research. doi: <a href="https://doi.org/10.1093/nar/gkad890" target="_blank"> 10.1093/nar/gkad890</a></i><br>
      <i><b>*</b> Authors contributed equally</i><br>
      Cite with
      <a href="{{ url_for('download_endnote') }}">EndNote</a> or
      <a href="{{ url_for('download_bibtex') }}">BibTeX</a>
    </p>
  </div>
</div>

  <!-- CasID Definitions -->
  <div id="cas_ids" class="summary">
  <!-- column 1 -->
  <div class="abstract">
    <div class="column-container">
      <div id="section_nuclease">
      <h2><b>DEFINING CasIDs</b></h2>
        <p> The three digits of CasID represent <a href="#section_nuclease">nuclease activity</a>, <a href="#section_targeting">target requirement</a>, and <a href="#section_multiplx">guide RNA (gRNA) design</a>, respectively. Definitions for all CasID digits are provided below.</p>
        <p>To demonstrate CasIDs, consider SpyCas9a, with a CasID of <b>1.1.1.</b></p>
            <ul class="ul-ul">
              <li>The <b>Nuclease Activity</b> for this enzyme falls under category 1, implying it has blunt double-strand cis nuclease-activity, and no trans nuclease-activity.<sup><a href="https://doi.org/10.1016/j.molcel.2018.11.021" target="_blank">1</a></sup>.</li>
              <li>The <b>Targeting Requirements</b> for SpyCas9a fit category 1, meaning it requires a 3' protospacer-adjacent motif (PAM)</li>
              <li>Finally, the <b>Guide RNA (gRNA) Design and Multiplexing</b> properties for this enzyme fall under category 1, such that the native CRISPR array for SpyCas9a requires a CRISPR RNA (crRNA) for targeting + a trans-acting crRNA (tracrRNA) for multiplexing to the protein backbone of the effector + additional factors for processing the array into mature guides. SpyCas9a can also be engineered to utilize a minimal array containing different single-guide RNAs (sgRNAs), represented as a contiguous, all-in-one crRNA + tracrRNA sequence.<sup><a href="https://doi.org/10.1038/nature09886" target="_blank">2,</a></sup><sup><a href="https://doi.org/10.1146/annurev-biophys-062215-010822" target="_blank">3</a></sup>.</li>
            </ul>
          <img src="{{url_for('static', filename='img/spycas9_example.svg')}}" alt="SpyCas9a Example">
        <h2><b>Primary Nuclease Activity</b></h2>
        <hr>
      </div>
        <h3>Nuclease activity is reported as the first digit in the CasID. Enzymes from the Class 2 Cas superfamily have diverse nuclease activities, both in an RNA-guided manner on a targeted sequence (in cis-) and sometimes on non-targeted nucleic acids upon binding (in trans-). The numerical annotation for nuclease activity relates to the known nucleic acid substrates for both the cis- and trans- activities of these enzymes. DNA sequences are shown in red and blue. RNA sequences (non-gRNA) are shown in green. Cleavage sites are shown with black triangles (▲▼). RNA guided Cas enzymes are shown in a fixed orientation to the nucleic acid substrate and not the value inferred by the Target Requirement.</h3>
        <img src="{{url_for('static', filename='img/nuc_activity.svg')}}" alt="Image 1">
    </div>
  </div>
  <!-- column 2 -->
  <div class="abstract">
    <div class="column-container">
      <div id="section_targeting">
        <h2><b>Target Requirement</b></h2>
        <hr>
      </div>
        <h3>Target requirement is reported as the second digit in the CasID. Target requirement refers to the orientation of the Cas enzyme relative to its RNA-guided target substrate, as dictated by its protospacer adjacent motif (PAM), for DNA substrates, and protospacer flanking site (PFS), for RNA. Briefly, a PAM is an adjacent sequence required for targeting DNA and a PFS is a neighboring sequence that suppresses targeting activity. The numerical annotation shows the orientation of the PAM, or PFS, relative to the protospacer and dictates the orientation of the ribonucleoprotein (RNP) relative to its targeted substrate. DNA sequences are shown in red and blue. RNA sequences (non-gRNA) are shown in green. PAMs (required adjacent sequences) are shown with a star (★) and PFSs (excluded adjacent sequences) are shown with an X.</h3>
        <img src="{{url_for('static', filename='img/targeting.png')}}" alt="Image 2">
    </div>
  </div>
  <!-- column 3 -->
  <div class="abstract">
    <div class="column-container">
      <div id="section_multiplex">
        <h2><b>Guide RNA (gRNA) design + Multiplexing</b></h2>
        <hr>
      </div>
        <h3>Guide RNA (gRNA) design and multiplexing ability is reported as the third digit in the CasID. Guide RNA (gRNA) design illustrates how the properties of the native CRISPR array inform the creation of a synthetic gRNA scaffold. The numerical annotation describes the relevant guide requirements in several contexts: within the native CRISPR (top), the minimal synthetic guide design for multiplexing (middle), and the sgRNA design for multiplexing (bottom). Spacers are represented by purple, green, and blue lines. CRISPR repeats that are incorporated in mature gRNAs are colored in orange and black if they’re lost during crRNA processing. Trans-acting RNAs (tracrRNA) are colored in red. Synthetically-added sequences (i.e. tetraloops) are shown in gray. </h3>
        <img src="{{url_for('static', filename='img/multiplex.png')}}" alt="Image 3">
    </div>
  </div>
</div>

  {% endblock %}
